modules/astparser/src/ast/declarations/ASTVariableDeclaration.h

```diff
@@ -54,11 +54,27 @@ class ASTVariableDeclaration : public ASTDeclaration {
     ASTName name;
 
 public:
+    ASTVariableDeclaration(ASTModifiers mods, ASTName&& tn, ASTName&& n) :
+        ASTDeclaration(),
+        modifiers(mods),
+        typeName(tn),
+        name(n) {}
```

Comment:
missing std::move - actually on many places here you are using r-value, but then you pass the value to the ctor of the argument - it won't call move ctor this way, you have to explicitly move the variable:
```Foo(Arg&& bar) : m_bar(std::move(bar))```